src/rules/getTypeExports.ts

```diff
@@ -25,27 +38,40 @@ function isType(
  * all of the exported Types/Interfaces. The exported interfaces and types will
  * be stored in a cache.
  */
-function parseTSTreeForExportedTypes(cacheKey: string, content: string): void {
+function parseTSTreeForExportedTypes(
+  cacheKey: string,
+  content: string,
+  context: LintContext,
+): void {
   const typeList: string[] = [];
-
   try {
     if (fileCache && fileCache.get(cacheKey)) {
       const ast = parse(content, { sourceType: 'module' });
       const cache = fileCache.get(cacheKey);
-
       if (!cache) {
         return;
       }
 
       ast.body.forEach((node: TSESTree.Statement) => {
         if (node.type === 'ExportNamedDeclaration') {
+          let typeExports: Set<string> | undefined = new Set();
+          const { source } = node;
+          const sourceName =
+            source && 'value' in source ? source.value : undefined;
+          if (typeof sourceName === 'string') {
+            // eslint-disable-next-line @typescript-eslint/no-use-before-define
+            typeExports = parseFileForTypedExports(sourceName, context);
+          }
           const { declaration, specifiers } = node;
 
           specifiers.forEach(specifier => {
-            if (specifier.local.name === specifier.exported.name) {
-              cache.add(specifier.local.name);
-            } else {
-              cache.add(`${specifier.local.name} as ${specifier.exported.name}`)
+            let aliasName = specifier.local.name;
+            if (specifier.local.name !== specifier.exported.name) {
+              aliasName = `${aliasName} as ${specifier.exported.name}`;
+            }
+            cache.add(aliasName);
+            if (typeExports && typeExports.has(specifier.local.name)) {
+              typeList.push(aliasName);
             }
```

Comment:
As per https://github.com/intuit/eslint-plugin-no-explicit-type-exports/pull/34/files#r685735909

If `bar.ts` looks like:
```typescript
interface Bar { test: string }
export {Bar}
```
and if an intermediate file like `barCollection.ts` looks like this:
```typescript
export { Bar } from './bar'
```
And in another level up you do:
```typescript
export { Bar } from './barCollection'
```